File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/fixed-asset/assets-maintenance/fault-update/fixed-asset-fault-history.model.ts
import * as moment from "moment";

export class FixedAssetFaultHistoryModel {
    // pk  
    public FaultHistoryId: number = 0;
    public FixedAssetStockId: number = 0;

    public FaultDate: string = moment().format("YYYY-MM-DD");;
    public FaultDescription: string;

    public FaultResolvedDate: string = moment().format("YYYY-MM-DD");;
    public FaultResolvedRemarks: string;

    public CreatedBy: number;
    public CreatedOn: string;
    public ModefiedBy: number;
    public ModefiedOn: string;
    public CreatedName: string;

    public IsFaultResolved: boolean = false;


}

File: Code/Components/DanpheEMR.ServerModel/FixedAssetModels/FixedAssetFaultHistoryModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class FixedAssetFaultHistoryModel
    {
        [Key]
        public int FaultHistoryId { get; set; }
        public int FixedAssetStockId { get; set; }

        public DateTime? FaultDate { get; set; }
        public string FaultDescription { get; set; }

        public DateTime? FaultResolvedDate { get; set; }
        public string FaultResolvedRemarks { get; set; }

        public DateTime? CreatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public bool IsFaultResolved { get; set; }







    }
}
